pkg/controller/replica.go

```diff
@@ -165,6 +165,7 @@ func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 				},
 				true,
 			)
+			r.controllerSharedState.tryCreateFailedReplicas(ctx, replica, r)
```

Comment:
Shouldn't this be done in the same function queued to the operation queue after the call to `cleanUpAzVolumeAttachmentByVolume` is called? In its current location, it will attempt to create new replicas before or concurrently with the attachment clean up and it's not likely any additional attachment would be available.

Reply:
@sunpa93 I wonder if the call to `tryCreateFailedReplicas` should be made in `cleanUpAzVolumeAttachmentByVolume`?

Reply:
`cleanUpAzVolumeAttachmentByVolume` also doesn't wait for the CRI to be deleted so we would end up with thes ame problem. What I was thinking was that we should start a goroutine after the cleanup is added to the operation queue to wait for all the attachments for the volume to be deleted and call `tryCreateFailedReplicas` once the wait succeeds.